src/main/java/com/google/sps/servlets/TagServlet.java

```diff
@@ -48,14 +50,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String[] tagIds = request.getParameterValues("tagIds");
     String getHidden = request.getParameter("getHidden");
 
+    // Only relevant in the case of returning a user's followed tags.
+    String token = request.getParameter("token");
+
     List<Tag> tags;
 
     if (tagIds == null) {
-      tags = db.getAllTags(getHidden != null && getHidden.equals("true"));
+      if (!(token == null || token.isEmpty())) {
+        String uid = Auth.getUid(token, response);
+        if (uid == null) {
+          return;
+        }
+        List<String> followedTags = db.followedTagIds(uid);
+        if (followedTags == null) {
+          response.setStatus(HttpServletResponse.SC_NO_CONTENT);
+          return;
+        }
+        tags = db.getTagsMatchingIds(followedTags);
+      } else {
+        tags = db.getAllTags(getHidden != null && getHidden.equals("true"));
+      }
```

Comment:
I the code involved with getting the tags followed by the user should be moved to another function that returns the resulting ArrayList<String>, for clarity.

Reply:
Gotcha